### PR TITLE
[FVM] Refactoring TransactionProcedure trace span creation

### PR DESCRIPTION
**Pull request by Tonix517**
- Adding `IsSampled()` to TransactionProcedure. Taking the message from [computer.go](https://github.com/onflow/flow-go/blob/master/engine/execution/computation/computer/computer.go#L411-L413) to `TransactionProcedure` to make code slightly easier to read.
- Isolating boilerplace code for creating tracer span for TransactionProcedure into a function.

**Test**
- [X] all fvm UT

### CODE REVIEW

File: fvm/transactionInvoker.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func (i TransactionInvoker) Process(
 
 	txIDStr := proc.ID.String()
 	var span otelTrace.Span
-	if ctx.Tracer != nil && proc.TraceSpan != nil {
```
Comment (pattyshack) [inline fvm/transactionInvoker.go:38]: This boilerplate (this + the next line) is very similar to environment.Tracer.StartSpanFromRoot (the "root" used by the environment is the span that's being created here).  We can probably refactor / reuse parts of StartSpanFromRoot here.

Comment (Tonix517) [reply fvm/transactionInvoker.go:38]: good suggestion. I'm keeping `IsSampled()` which I need for my next PR to emit computation effort vector metrics.

File: fvm/transaction.go [hunk 2]
```diff
@@ -48,6 +50,18 @@ func (proc *TransactionProcedure) SetTraceSpan(traceSpan otelTrace.Span) {
 	proc.TraceSpan = traceSpan
 }
 
+func (proc *TransactionProcedure) IsSampled() bool {
+	return proc.TraceSpan != nil
+}
+
+func (proc *TransactionProcedure) StartSpanFromProcTraceSpan(
+	tracer module.Tracer, spanName trace.SpanName) otelTrace.Span {
```
Comment (pattyshack) [inline fvm/transaction.go:58]: add extra newlines to clarify the method signature

**Review by pattyshack — APPROVED** [review]

File: fvm/transaction.go (hunk 2 above)
Comment (Tonix517) [reply fvm/transaction.go]: addressed

**Review by janezpodhostnik — APPROVED** [review]

**Comment by Tonix517** [comment]
bors merge

**Pull request merged**